### ISSUE TITLE
Issue during fee estimation

### ISSUE THREAD

**Issue by mrMateosantana** [software-mansion/starknet.swift#144]
### What happened

Error while trying to estimate fees for transaction
`            let feeEstimate = try await testStarknetAccount.estimateFeeV3(calls: calls, nonce: nonce)
`
Probably failing somewhere here: 
` func estimateFee(for transaction: any StarknetExecutableTransaction, simulationFlags: Set<StarknetSimulationFlagForEstimateFee>) async throws -> StarknetFeeEstimate {
        try await estimateFee(for: [transaction], simulationFlags: simulationFlags)[0]
    }
`

### Stack trace

jsonRpcError(-32602, "Invalid params", Optional("{\"reason\":\"unknown field `simulation_flags`, expected `request` or `block_id` at line 1 column 856\"}"))

### Steps to reproduce

Trying to estimate fees for a v3 transaction

### SDK Version

v0.8.0

### Language version

5.9

### Is there an existing issue for this?

- [X] I have searched the existing issues and verified no issue exits for this problem.

**Comment by DelevoXDG** [comment]
Hi @mrMateosantana 
Can you please tell me what is the version of the RPC node you're using? You can verify this by running `StarknetProvider.specVersion()`

**Comment by mrMateosantana** [comment]
> Hi @mrMateosantana Can you please tell me what is the version of the RPC node you're using? You can verify this by running `StarknetProvider.specVersion()`

![image](https://github.com/software-mansion/starknet.swift/assets/157704598/fde4bff3-8be4-40a2-9862-cc82e4c0b5fc)
Version "0.5.1", I guess then maybe those flags are only supported on RPC v6 right? Do you know recommend any provider, because seems that alchemy and infura still on v5

**Comment by DelevoXDG** [comment]
That is correct. You can checkout [Blast](https://blastapi.io/public-api/starknet) public RPC endpoints.

Alternatively, you can use starknet.swift 0.7.4 that is compatible with RPC 0.5.1

**Comment by mrMateosantana** [comment]
Thanks, that was it. With blast v0.6 now the fee estimation mechanism is properly working.

**Comment by DelevoXDG** [comment]
Glad it helped!
